spi/common/json-ld-spi/src/main/java/org/eclipse/edc/jsonld/spi/JsonLd.java

```diff
@@ -38,15 +39,54 @@ public interface JsonLd {
      * @param json the expanded json.
      * @return a successful {@link Result} containing the compacted {@link JsonObject} if the operation succeed, a failed one otherwise
      */
-    Result<JsonObject> compact(JsonObject json);
+    default Result<JsonObject> compact(JsonObject json) {
+        return compact(json, DEFAULT_SCOPE);
+    }
 
     /**
-     * Register a JsonLD namespace
+     * Compact a JsonLD document. The context will be generated from registered contexts and namespaces.
+     *
+     * @param json  the expanded json.
+     * @param scope the scope to apply during the compaction process
+     * @return a successful {@link Result} containing the compacted {@link JsonObject} if the operation succeed, a failed one otherwise
+     */
+    Result<JsonObject> compact(JsonObject json, String scope);
+
+    /**
+     * Register a JsonLD namespace in the default scope
+     *
+     * @param prefix     the prefix
+     * @param contextIri the string representing the IRI where the context is located
+     */
+    default void registerNamespace(String prefix, String contextIri) {
+        registerNamespace(prefix, contextIri, DEFAULT_SCOPE);
+    }
+
+    /**
+     * Register a JsonLD namespace in a provided scope
      *
-     * @param prefix the prefix
+     * @param prefix     the prefix
      * @param contextIri the string representing the IRI where the context is located
+     * @param scope      the scope where the prefix will be applied
      */
-    void registerNamespace(String prefix, String contextIri);
+    void registerNamespace(String prefix, String contextIri, String scope);
```

Comment:
Maybe I would put "scope" as first parameter, like it was done in the policy engine

Reply:
We have the other signature without the scope and it will lead to different
arg positions which could confuse people?

```Java
 void registerNamespace(String prefix, String contextIri)
 void registerNamespace(String scope. String prefix, String contextIri)
```

Reply:
mmh.. yes, it could be as they are all strings... ok for me to leave as it is